lib/livebook_web/live/hub/edit/team_component.ex

```diff
@@ -552,21 +557,44 @@ defmodule LivebookWeb.Hub.Edit.TeamComponent do
       """
 
     zta = zta_env(socket.assigns.zta)
-    dockerfile = if zta, do: base <> zta <> apps, else: base <> apps
+
+    dockerfile =
+      for block <- [secrets, file_systems, zta],
+          not is_nil(block),
+          reduce: base,
+          do: (acc -> acc <> block)
+
+    dockerfile = dockerfile <> apps
```

Comment:
```suggestion
    dockerfile =
      [secrets, file_systems, zta, apps]
      |> Enum.reject(&is_nil/1)
      |> Enum.join()
```

Reply:
(Btw inside `for` instead of `not is_nil(x)`, you can just use `x` as the filter :))